src/pages/dashboard/Explore.tsx

```diff
@@ -85,101 +92,85 @@ const StyledResultListHeaderWrapper = styled.div`
-      const offset = pageTerm && Number(pageTerm) !== firstPage ? (Number(pageTerm) - 1) * NumberOfResultsPrPage : 0;
-      return {
-        ...emptyQueryObject,
-        query: searchTerms.get(SearchParameters.query) ?? '',
-        offset: offset,
-        limit: NumberOfResultsPrPage,
-        resourceTypes: resourceTypes,
-        institutions: institutions,
-        licenses: licenses,
-        tags: tags,
-        queryFromURL: true,
-        allowSearch: true,
-        showInaccessible: showInaccessible,
-      };
-    };
-    if (!queryObject.queryFromURL && !queryObject.allowSearch) {
-      setQueryObject(createQueryFromUrl());
+  const triggerSearch = async (queryObject: QueryObject) => {
+    try {
+      setIsSearching(true);
+      const response = await searchResources(queryObject);
+      setSearchError(undefined);
+      setSearchResult(response.data);
+      setResources(response.data.resourcesAsJson.map((resourceAsString: string) => JSON.parse(resourceAsString)));
+    } catch (error) {
+      setSearchError(error);
+    } finally {
+      setIsSearching(false);
     }
-  }, [location, queryObject.allowSearch, queryObject.queryFromURL]);
+  };
+
+  useEffect(() => {
+    triggerSearch(queryObject);
+  }, [queryObject]);
 
   useEffect(() => {
-    const reWriteUrl = () => {
-      let url = `?`;
-      if (queryObject.query.length > 0) url += `${SearchParameters.query}=${queryObject.query}`;
-      if (queryObject.institutions.length > 0)
-        url += queryObject.institutions
-          .map((institution) => `&${SearchParameters.institution}=${institution}`)
-          .join('');
-      if (queryObject.resourceTypes.length > 0)
-        url += queryObject.resourceTypes.map((type) => `&${SearchParameters.resourceType}=${type}`).join('');
-      if (queryObject.licenses.length > 0)
-        url += queryObject.licenses.map((licenseCode) => `&${SearchParameters.license}=${licenseCode}`).join('');
-      if (queryObject.tags.length > 0) url += queryObject.tags.map((tag) => `&${SearchParameters.tag}=${tag}`).join('');
-      if (queryObject.showInaccessible) url += `&${SearchParameters.showInaccessible}=true`;
-      history.push(url);
-    };
-
-    const triggerSearch = async () => {
-      if (!queryObject.queryFromURL) {
-        reWriteUrl();
-      }
-      if (queryObject) {
-        try {
-          setIsSearching(true);
-          const response = await searchResources(queryObject);
-          setSearchError(undefined);
-          setSearchResult(response.data);
-          setResources(response.data.resourcesAsJson.map((resourceAsString: string) => JSON.parse(resourceAsString)));
-        } catch (error) {
-          setSearchError(error);
-        } finally {
-          setIsSearching(false);
-        }
-      }
-    };
-    if (queryObject.allowSearch) triggerSearch();
-    if (queryObject.offset === 0) setPage(firstPage);
-  }, [queryObject, history]);
+    if (!attached) {
```

Comment:
hva står atteched for ? Kunne du gitt den et navn som forklarer intensjon ?

Reply:
Uten attached så fikk jeg en haug popstate-lyttere som satt å kjørte søk på nytt igjen alle sammen. Jo flere ganger man klikket rundt på siden jo flere popstate-lyttere fikk man. Og hvis man faktisk klikket på bakover knappen eller fremover knappen så fikk man enda flere popstate lyttere (queryObject endrer seg, så lag ny popstate lytter), som da havnet i en uendelig løkke

Reply:
joda. men savner et mer forklarende ord. hasAddedPopstateListener f.ex ?

Reply:
Ahh, sånn sett. Skal ordne

Reply:
Fikset